File: Controllers/ClockProps.cs
namespace time.Controllers;

public class ClockProps {
    public Guid Id { get; set; }
    public string FontFamily {get; set;} = "courier";
    public int[] AvailableFontSizes {get; }  = new[] { 12, 24, 48, 64 };
    public int TitleFontSize {get; set;} = 64;
    public int ClockFontSize {get ; set;} = 48;
    public bool BlinkColons {get; set;} = true;
    public string FontColor {get; set;} = "black";
    public ClockProps()
    {
        Id = Guid.NewGuid();
    }
}

File: ClientApp/src/components/ClockProps.js
export default class ClockProps {
  constructor() {
    this.fontFamily = 'courier'
    this.availableFontSizes = [12, 24, 48, 64]
    this.titleFontSize = 64
    this.clockFontSize = 48
    this.blinkColons = true
    this.fontColor = 'black'
    this.Timer = ""
    }

    updateTimer() {
        const currentDate = new Date();
        const year = currentDate.getFullYear();
        const month = (currentDate.getMonth() + 1).toString().padStart(2, '0');
        const day = currentDate.getDate().toString().padStart(2, '0');
        const hours = currentDate.getHours().toString().padStart(2, '0');
        const minutes = currentDate.getMinutes().toString().padStart(2, '0');
        const seconds = currentDate.getSeconds().toString().padStart(2, '0');
        this.Timer = `${year}-${month}-${day}T${hours}:${minutes}:${seconds}`;
    }
}
